From ebc7cf3152b8c4d64779c64fd3f5d3d6a25166b9 Mon Sep 17 00:00:00 2001
From: Todd Whiteman <todd.whiteman@joyent.com>
Date: Mon, 28 May 2018 15:41:24 -0700
Subject: [PATCH] TRITON-434 volapi crashing with "Cannot read property 'push'
 of undefined" Reviewed by: Josh Wilsdon <josh@wilsdon.ca> Approved by: Josh
 Wilsdon <josh@wilsdon.ca>

---
 package.json      | 2 +-
 volapi-updater.js | 7 +++----
 2 files changed, 4 insertions(+), 5 deletions(-)

diff --git a/package.json b/package.json
index 7130981..e928088 100644
--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "sdc-volapi",
-  "version": "1.0.0",
+  "version": "1.0.1",
   "description": "SDC's Volumes API",
   "main": "server.js",
   "scripts": {
diff --git a/volapi-updater.js b/volapi-updater.js
index d11197e..1fc6cea 100644
--- a/volapi-updater.js
+++ b/volapi-updater.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2017, Joyent, Inc.
+ * Copyright (c) 2018, Joyent, Inc.
  */
 
 /*
@@ -1063,11 +1063,10 @@ function updateReferencesAndReservationsForVm(vmUuid, options, callback) {
 
                     mod_assert.optionalArrayOfUuid(volumeValue.refs,
                         'volumeValue.refs');
-                    volumeRefs = volumeValue.refs;
+                    volumeRefs = volumeValue.refs || [];
 
                     if (ctx.shouldAddReferences) {
-                        if (volumeRefs === undefined || volumeRefs === null ||
-                            volumeRefs.indexOf(vmUuid) === -1) {
+                        if (volumeRefs.indexOf(vmUuid) === -1) {
                             volumeRefs.push(vmUuid);
                             volumeObjectsToUpdate.push(volumeObj);
                         }
-- 
2.21.0

